test/components/fetchers/test_serpex.py

```diff
@@ -0,0 +1,271 @@
+            with pytest.raises(httpx.HTTPStatusError):
+                fetcher.run(query="test query")
+
+    def test_run_with_network_error(self):
+        """Test run method with network error"""
+        with patch("haystack.components.fetchers.serpex.httpx.Client.get") as mock_get:
+            mock_get.side_effect = httpx.RequestError("Connection failed", request=Mock())
+
+            fetcher = SerpexWebSearch(api_key="test-api-key")
+            with pytest.raises(httpx.RequestError):
+                fetcher.run(query="test query")
+
+    def test_run_verifies_api_key_in_headers(self, mock_serpex_response):
+        """Test that API key is properly included in request headers"""
+        with patch("haystack.components.fetchers.serpex.httpx.Client.get") as mock_get:
+            mock_response = Mock(status_code=200)
+            mock_response.json.return_value = mock_serpex_response
+            mock_get.return_value = mock_response
+
+            fetcher = SerpexWebSearch(api_key="secret-key-123")
+            fetcher.run(query="test query")
+
+            # Verify the Authorization header was set correctly
+            call_args = mock_get.call_args
+            assert call_args[1]["headers"]["Authorization"] == "Bearer secret-key-123"
+            assert call_args[1]["headers"]["Content-Type"] == "application/json"
+
+    def test_run_uses_correct_api_endpoint(self, mock_serpex_response):
+        """Test that the correct SERPEX API endpoint is used"""
+        with patch("haystack.components.fetchers.serpex.httpx.Client.get") as mock_get:
+            mock_response = Mock(status_code=200)
+            mock_response.json.return_value = mock_serpex_response
+            mock_get.return_value = mock_response
+
+            fetcher = SerpexWebSearch(api_key="test-api-key")
+            fetcher.run(query="test query")
+
+            # Verify the correct endpoint was called
+            call_args = mock_get.call_args
+            assert call_args[0][0] == "https://api.serpex.dev/api/search"
+
+    @pytest.mark.integration
+    @pytest.mark.skipif(not os.environ.get("SERPEX_API_KEY"), reason="SERPEX_API_KEY not set")
+    def test_run_with_real_api(self):
+        """Integration test with real SERPEX API"""
+        api_key = os.environ.get("SERPEX_API_KEY")
+        fetcher = SerpexWebSearch(api_key=api_key)
+
+        result = fetcher.run(query="Haystack LLM framework")
+        documents = result["documents"]
+
+        # Basic assertions
+        assert len(documents) > 0
+        assert all(isinstance(doc, Document) for doc in documents)
+        assert all("title" in doc.meta for doc in documents)
+        assert all("url" in doc.meta for doc in documents)
+        assert all(doc.content for doc in documents)
+
+    @pytest.mark.integration
+    @pytest.mark.skipif(not os.environ.get("SERPEX_API_KEY"), reason="SERPEX_API_KEY not set")
+    def test_run_with_different_engines(self):
```

Comment:
Just wanted to ask if you were able to run these integration tests and check that they pass?